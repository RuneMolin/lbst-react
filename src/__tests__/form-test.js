import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

import Form from '../form'

describe('The form', () => {
  it('initially looks as expected', () => {
    const subject = shallow(<Form />)
    expect(subject).toMatchSnapshot()
  })

  it('displays the "navn" field as success when anything is entered', () => {
    const subject = shallow(<Form />)

    console.info(subject.find('input[name="navn"]').prop('class'))
    //expect(subject.find('input[name="navn"]'))
  })
})
