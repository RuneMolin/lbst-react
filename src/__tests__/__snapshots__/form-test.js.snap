// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form initially looks as expected 1`] = `
<form
  className="box"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="navn"
    >
      Navn
    </label>
    <div
      className="control has-icons-left"
    >
      <input
        className="input is-danger"
        defaultValue=""
        name="navn"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fa fa-user"
        />
      </span>
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="tlf"
    >
      Telefonnummer
    </label>
    <div
      className="control has-icons-left"
    >
      <input
        className="input is-danger"
        defaultValue=""
        name="tlf"
        onChange={[Function]}
        required={true}
        type="tel"
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fa fa-phone"
        />
      </span>
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <label
        className="checkbox"
      >
        <input
          checked={false}
          name="jegErEnig"
          onChange={[Function]}
          required={true}
          type="checkbox"
        />
        Jeg lyver ikke
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <button
      className="button is-primary"
      disabled={true}
      type="submit"
    >
      OK
    </button>
  </div>
</form>
`;
